scripts/src/main/resources/scripts/functions

```diff
@@ -1810,53 +1816,138 @@ function doGetSoftwareVersion() {
   fi
 }
 
+# $1: software
+function doGetSoftwareEdition() {
+  local software="${1}"
+  local editionVar
+  editionVar="$(echo "${software/-/_}" | awk '{print toupper($0)}')_EDITION"
+  local edition
+  edition="${!editionVar}"
+  if [ -z "${edition}" ]
+  then
+    edition="${software}"
+  fi
+  echo "${edition}"
+}
+
+# $1: software
+# $2: optional prefix
+function doGetLatestSoftwareVersion() {
+  local software="${1}"
+  local prefix="${2}"
+  local edition
+  local version
+  doUpdateUrls
+  edition=$(doGetSoftwareEdition "${software}")
+  if [ -z "${prefix}" ]
+  then
+    version="$(find "${DEVON_IDE_HOME}"/urls/"${software}"/"${edition}" -mindepth 1 -maxdepth 1 -print | awk -F'/' '{print $NF}' | sort -rV | head -1)"
+  else
+    prefix="${prefix:0:${#prefix}-1}"
+    prefix="${prefix/./[.]}"
+    version="$(find "${DEVON_IDE_HOME}"/urls/"${software}"/"${edition}" -mindepth 1 -maxdepth 1 -print | awk -F'/' '{print $NF}' | grep "^${prefix}" | sort -rV | head -1)"
+  fi
+  echo "${version}"
+} 
+
 # $1: software
 # $2: selected version
 # $3: commandlet name
 function doSetSoftwareVersion() {
   local software="${1}"
   local version="${2}"
   local commandlet="${3}"
-  if [ -n "${version}" ]
+  local edition
+  if [ -d "${DEVON_IDE_HOME}/urls" ]
   then
-    if [ "${version}" = "latest" ]
+    if [ -n "${version}" ]
     then
-      doUpdateMirrors
-      version="$(head -n 1 "${DEVON_IDE_HOME}/mirrors/${software}/available-versions")"
+      if [ "${version}" = "latest" ]
+      then
+        doUpdateUrls
```

Comment:
this already happens in `doGetLatestSoftwareVersion`. Doing `git pull` twice is pointless waste. Please remove this line.

Reply:
Yes, thanks for the hint. Line is removed.